### PR TITLE
[OCPCLOUD-921] add alert for machine with long deletion phase

### PR DESCRIPTION
**Pull request by elmiko**
This change adds the alert for a machine with phase "Deleting" for
longer than 180 minutes, at "warning" severity. Also updates the alerts
documentation with new information.

The MachineWithNoRunningPhase alert has also been modified to exclude
machines with the "Deleting" phase to ensure that we do not double fire
an alert for this condition.

ref: https://issues.redhat.com/browse/OCPCLOUD-921

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
I think this looks fine, would be good to get an SRE opinion on it before we merge though

**Comment by RiRa12621** [comment]
@JoelSpeed LGTM from our end.
[x] Documentation exists
[x] Root cause based alert does not exceed warning level

**Comment by RiRa12621** [comment]
/assign @RiRa12621 
assign to myself so it's a bit easier for me find this later

**Comment by JoelSpeed** [comment]
/approve
/retest

Thanks for the review @RiRa12621

**Comment by elmiko** [comment]
just realized i set this for 3hours, but we specified 6hours in the enhancement. i'm going to bump it up.

File: docs/user/Alerts.md
Comment (michaelgugino) [inline docs/user/Alerts.md:45]: Add: A pod with a very long GracePeriodTimeout has not stopped yet

File: docs/user/Alerts.md
Comment (michaelgugino) [inline docs/user/Alerts.md:51]: We should work to ensure in the future that looking at the logs is not a necessity.  We should prioritize putting the information on the machine's status.

**Review by michaelgugino — COMMENTED** [review]
A couple small things, otherwise looks good to me.

**Comment by elmiko** [comment]
/hold
i need to fix up a few thing

**Comment by 2uasimojo** [comment]
👀

**Comment by elmiko** [comment]
updated the pr:
* changed error name to MachineNotYetDeleted
* include a mention of graceful termination period in possible causes
* update alert timing to 360 minutes

**Comment by dofinn** [comment]
The alert itself LGTM
@RiRa12621 @jeremyeder is this a candidate for https://github.com/openshift/runbooks ?

**Comment by RiRa12621** [comment]
Didn't see you ping @dofinn 
yes it is, but we're not super clear yet on the structure. I'll port this one over once we're ready with the runbooks repo

**Comment by JoelSpeed** [comment]
/retest

@dofinn Are you happy for us to merge this into 4.8 or will that cause issues for you this late?

**Comment by RiRa12621** [comment]
lgtm from my end @JoelSpeed 
as for the runbooks dom mentioned, we're still bootstrapping that bit, so don't worry about it for now

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-workers-rhel7

This job no longer exists and so will never be able to pass

**Review by michaelgugino — APPROVED** [review]

**Review by michaelgugino — APPROVED** [review]
/approve

**Comment by JoelSpeed** [comment]
This is a feature, so we need to wait until master opens before we can merge it. Discussed already with @elmiko, if we leave this without hold, it should merge as soon as master reopens

**Comment by RiRa12621** [comment]
Are we still confident this will happen automatically?

**Comment by elmiko** [comment]
> Are we still confident this will happen automatically?

i want to believe that, but if not we will kick the CI machine to merge it ;)

**Comment by RiRa12621** [comment]
Sounds perfect. Was just wondering since it's been a month an I wasn't sure if master had opened meanwhile.

**Comment by elmiko** [comment]
it looks like 4.9 development will open on June 11.

**Comment by stbenjam** [comment]
/retitle Bug 1974758: [OCPCLOUD-921] add alert for machine with long deletion phase

**Pull request merged**
